[run-webkit-tests] Surpress exception when terminating settings app

JonWBedard opened the pull request:
#### d6f0fede47d017f957b83b47e0bbe53464f39a20
<pre>
[run-webkit-tests] Surpress exception when terminating settings app
<a href="https://bugs.webkit.org/show_bug.cgi?id=260860">https://bugs.webkit.org/show_bug.cgi?id=260860</a>
rdar://114626133

Reviewed by Ryan Haddad.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDevice.is_usable): Combine exit codes from starting and stopping Settings app.

Canonical link: <a href="https://commits.webkit.org/267428@main">https://commits.webkit.org/267428@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9b1cb911e93de697a0a11e7d18aa7d17077821bf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16497 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16819 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17258 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18276 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15471 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/16687 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20050 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16962 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17823 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16694 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17109 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14267 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19053 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/16622 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14353 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14951 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21749 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15341 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15117 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/19433 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15711 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13342 "1 failures") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14912 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19281 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2038 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15537 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

robert-jenner reviewed (approved):
This only had failures on [macOS-AppleSilicon-Ventura-Debug-WK2-Tests-EWS](https://ews-build.webkit.org/#/builders/17) and none of the failures mentioned there were introduced with this patch. Further, I think this should resolve our issues. Approving this.

JonWBedard committed:
[run-webkit-tests] Surpress exception when terminating settings app
https://bugs.webkit.org/show_bug.cgi?id=260860
rdar://114626133

Reviewed by Ryan Haddad.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDevice.is_usable): Combine exit codes from starting and stopping Settings app.

Canonical link: https://commits.webkit.org/267428@main
